core/src/test/java/com/linecorp/armeria/server/AnnotatedHttpServiceTest.java

```diff
@@ -423,7 +474,7 @@ public void testAnnotatedHttpService() throws Exception {
             }
             // Run case 3 but with not-mapped HTTP method (Post).
             try (CloseableHttpResponse res = hc.execute(new HttpPost(rule.httpUri("/1/string/blah")))) {
-                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 405 Method Not Allowed"));
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 404 Not Found"));
```

Comment:
It's hard to distinguish 405 from 404 in the current implementation because we need to find the next matching when the method is not matched. 
But, anyway it may be possible. However I'm also curious whether 405 is correct in this kind of cases or not because there's definitely no service instance to be able to handle the request.